--- a/src/utilities/memory.c
+++ b/src/utilities/memory.c
@@ -666,7 +666,9 @@ hypre_Memcpy_core(void *dst, void *src, size_t size, hypre_MemoryLocation loc_ds
       // hipMemcpy(DtoD) causes a host-side synchronization, unlike cudaMemcpy(DtoD),
       // use hipMemcpyAsync to get cuda's more performant behavior. For more info see:
       // https://github.com/mfem/mfem/pull/2780
-      HYPRE_HIP_CALL( hipMemcpyAsync(dst, src, size, hipMemcpyDeviceToDevice) );
+      // ======= PATCH: Async does not behave correctly on MI300 systems.
+      // =======        Reverting.
+      HYPRE_HIP_CALL( hipMemcpy(dst, src, size, hipMemcpyDeviceToDevice) );
 #endif
 
 #if defined(HYPRE_USING_SYCL)
@@ -797,7 +799,9 @@ hypre_Memcpy_core(void *dst, void *src, size_t size, hypre_MemoryLocation loc_ds
       // hipMemcpy(DtoD) causes a host-side synchronization, unlike cudaMemcpy(DtoD),
       // use hipMemcpyAsync to get cuda's more performant behavior. For more info see:
       // https://github.com/mfem/mfem/pull/2780
-      HYPRE_HIP_CALL( hipMemcpyAsync(dst, src, size, hipMemcpyDeviceToDevice) );
+      // ======= PATCH: Async does not behave correctly on MI300 systems.
+      // =======        Reverting.
+      HYPRE_HIP_CALL( hipMemcpy(dst, src, size, hipMemcpyDeviceToDevice) );
 #endif
 
 #if defined(HYPRE_USING_SYCL)
 